### PR TITLE
[New Rule] Security Software Discovery using Grep

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/742

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/discovery_security_software_grep.toml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+[metadata]
+creation_date = "2020/12/20"
+maturity = "production"
+updated_date = "2020/12/20"
+
+[rule]
+author = ["Elastic"]
+description = "Identifies the use of grep command to discover known security process names such as AntiVirus or Host Firewall."
```
Comment (peasead) [inline rules/macos/discovery_security_software_grep.toml:8]:
```suggestion
description = "Identifies the use of the grep command to discover known third-party macOS security tools, such as AntiVirus or Host Firewall."
```

**Review by peasead — APPROVED** [review]

**Review by threat-punter — CHANGES_REQUESTED** [review]
Is it possible to tweak this rule to include Linux hosts too?

**Comment by Samirbous** [comment]
> Is it possible to tweak this rule to include Linux hosts too?

good idea, we can do that, need to get some AV processes names for Linux hosts.

File: rules/cross-platform/discovery_security_software_grep.toml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+author = ["Elastic"]
+description = """
+This rule identifies the use of the grep command to discover known third-party macOS and Linux security tools, such as
+AntiVirus or Host Firewall.
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*", "auditbeat-*"]
+language = "kuery"
+license = "Elastic License"
+name = "Security Software Discovery using Grep"
```
Comment (bm11100) [inline rules/cross-platform/discovery_security_software_grep.toml:16]:
```suggestion
name = "Security Software Discovery via Grep"
```

File: rules/cross-platform/discovery_security_software_grep.toml [hunk 3]
```diff
@@ -0,0 +1,71 @@
+[metadata]
+creation_date = "2020/12/20"
+maturity = "production"
+updated_date = "2020/12/20"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule identifies the use of the grep command to discover known third-party macOS and Linux security tools, such as
+AntiVirus or Host Firewall.
```
Comment (bm11100) [inline rules/cross-platform/discovery_security_software_grep.toml:10]:
```suggestion
Identifies the use of the grep command to discover known third-party macOS and Linux security tools, such as
Antivirus or Host Firewall details.
```

File: rules/cross-platform/discovery_security_software_grep.toml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+license = "Elastic License"
+name = "Security Software Discovery using Grep"
+risk_score = 47
+rule_id = "870aecc0-cea4-4110-af3f-e02e9b373655"
+severity = "medium"
+tags = ["Elastic", "Host", "macOS", "Linux", "Threat Detection", "Discovery"]
+type = "query"
+
+query = '''
+event.category : process and event.type : (start or process_started) and
```
Comment (bm11100) [inline rules/cross-platform/discovery_security_software_grep.toml:24]: consider switching the query to EQL to take advantage of case insensitivity, i.e, -
```
process.args: 
  (
  "Little Snitch",
  "Avast*",
  "Avira*",
  "ESET*",
  etc...
  )
```

Comment (Samirbous) [reply rules/cross-platform/discovery_security_software_grep.toml:24]: case sensitivity not needed

**Review by threat-punter — APPROVED** [review]
👌

File: rules/cross-platform/discovery_security_software_grep.toml (hunk 4 above)
Comment (bm11100) [reply rules/cross-platform/discovery_security_software_grep.toml:24]: I'll disagree on that one. I think process.args should be case insensitive.  If an attacker executed `ps -ef | grep Elastic-Endpoint`, then it wouldn't trigger `elastic-endpoint` in the rule.

Comment (Samirbous) [reply rules/cross-platform/discovery_security_software_grep.toml:24]: there are tons of ways to bypass this detection, this is a light one, defenders can extend as they which.

**Pull request merged**